include/boost/gil/image_processing/scaling.hpp

```diff
@@ -5,6 +5,29 @@
 
 namespace boost{
 namespace gil{
+/// \defgroup ScalingAlgorithms
+/// \brief Algorthims suitable for rescaling
+///
+/// These algorithms are used to improve image
+/// quality after image resizing is made.
+
+/// \defgroup DownScalingAlgorithms
+/// \ingroup ScalingAlgorithms
+/// \brief Algorthims suitable for downscaling
+///
+/// These algorithms provide best results when used
+/// for downscaling. Using for upscaling will probably
+/// provide less than good results.
+
+
+/// \brief a single step of lanczos downscaling
+/// \ingroup DownScalingAlgorithms
+///
+/// Use this algorithm to scale down source image
+/// into a smaller image with reasonable quality.
+/// Do note that having a look at the output once
+/// is a good idea, since it might have ringing
+/// artifacts.
 template <typename ImageView>
 void lanczos_at(long int source_x, long int source_y, 
                 long int target_x, long int target_y, 
```

Comment:
coding style nit-pick: Slightly different alignment has been agreed for GIL:

https://github.com/BoostGSoC19/gil-olzhas/blob/a123653127d6619c3bd211f74114d9962b5a4bf3/include/boost/gil/algorithm.hpp#L137-L144

GIL function names can be longer (as preferred descriptive than cryptic), so ligning to the right has this disadvantage:

```
void my_super_duper_algorith_better_than_lanczos_at(long int source_x, long int source_y, 
                                                    long int target_x, long int target_y, 
```

Having big blank space on the left is not as friendly for left-to-right reading.

Reply:
Should I write one argument per line or concatenate if readable?

Reply:
It's all driven by line length limit:
If all fit in single line up to 90 characters limit, then prefer single line.
If single line would be longer than 90, then split one argument per line as shown earlier.

Reply:
Actually (since we are talking about coding style): In Mateusz' example above, I'd prefer the first argument to follow the opening paren without newline, and the other arguments to be vertically aligned with the first. That's just a matter of readability: avoid newlines if not needed to make a statement (or expression) fit into a line.

Reply:
@stefanseefeld I can't recall we've made a clear agreement on that, so there is no strict guideline. Or perhaps it was discussed as part of some of the many code reviews.
I just would really like to avoid the big empty multiline gaps, that are e.g. followed by 2-3 lines of function definition.

I'd rather increase the line length limit to 100 or 110 characters. Shall we?

Reply:
@simmplecoder You can go wider ;-)
https://github.com/boostorg/gil/commit/5024f6ffdddecf4c0432892bf77fc5cf2ae1c1f8